cylc/flow/data_store_mgr.py

```diff
@@ -539,19 +542,173 @@ def generate_definition_elements(self):
+            descendant=False, is_parent=False):
+        """Construct nodes/edges for children/parents of source node."""
+        for t_name, t_point, is_abs in items:
+            t_node = TaskID.get(t_name, t_point)
+            t_id = (
+                f'{self.workflow_id}{ID_DELIM}{t_point}{ID_DELIM}{t_name}')
+            # Initiate edge element.
+            if is_parent:
+                e_id = (
+                    f'{self.workflow_id}{ID_DELIM}{t_node}{ID_DELIM}{s_node}')
+            else:
+                e_id = (
+                    f'{self.workflow_id}{ID_DELIM}{s_node}{ID_DELIM}{t_node}')
+            if e_id in self.n_window_edges[active_id]:
+                continue
+            if (
+                e_id not in self.data[self.workflow_id][EDGES] and
+                e_id not in self.added[EDGES] and
+                edge_distance <= self.n_edge_distance
+            ):
+                self.added[EDGES][e_id] = PbEdge(
+                    id=e_id,
+                    source=s_id,
+                    target=t_id
+                )
+                # Add edge id to node field for resolver reference
+                self.updated[TASK_PROXIES].setdefault(
+                    t_id,
+                    PbTaskProxy(id=t_id)).edges.append(e_id)
+                self.updated[TASK_PROXIES].setdefault(
+                    s_id,
+                    PbTaskProxy(id=s_id)).edges.append(e_id)
+                self.n_window_edges[active_id].add(e_id)
+            if t_id in self.n_window_nodes[active_id]:
+                continue
+            self.increment_graph_window(
+                t_name, t_point, flow_label,
+                copy(edge_distance), active_id, descendant)
+
+    def remove_active_node(self, name, point):
+        """Flag removal of this active graph window."""
+        tp_id = f'{self.workflow_id}{ID_DELIM}{point}{ID_DELIM}{name}'
+        if tp_id in self.all_task_pool:
+            self.all_task_pool.remove(tp_id)
+        # flagged isolates/end-of-branch nodes for pruning on removal
+        if (
+                tp_id in self.prune_trigger_nodes and
+                tp_id in self.prune_trigger_nodes[tp_id]
+        ):
+            self.prune_flagged_nodes.update(self.prune_trigger_nodes[tp_id])
+            del self.prune_trigger_nodes[tp_id]
+            self.prune_pending = True
+
+    def add_runahead_node(self, name, point):
+        """Flag removal of this active graph window."""
+        tp_id = f'{self.workflow_id}{ID_DELIM}{point}{ID_DELIM}{name}'
+        self.all_task_pool.add(tp_id)
+
+    def generate_ghost_task(self, tp_id, name, point, flow_label):
         """Create task-point element populated with static data.
 
         Args:
-            t_id (str): data-store task ID.
             tp_id (str): data-store task proxy ID.
-            point_string (str): Valid cycle point string.
+            itask (cylc.flow.TaskProxy): Task proxy object.
```

Comment:
`itask` is not in this function? It's missing `name`, `point`, and `flow_label`.

Reply:
Yes, not anymore, oops